src/promptflow/promptflow/_sdk/_telemetry/activity.py

```diff
@@ -172,7 +176,14 @@ def wrapper(self, *args, **kwargs):
             # update activity name according to kwargs.
             _activity_name = update_activity_name(activity_name, kwargs=kwargs)
             with log_activity(logger, _activity_name, activity_type, custom_dimensions):
-                return f(self, *args, **kwargs)
+                try:
+                    return f(self, *args, **kwargs)
+                finally:
+                    if _activity_name in HINT_ACTIVITY_NAME:
+                        if sys.platform.startswith("win"):
+                            asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
+                        async_run_allowing_running_loop(check_latest_version)
```

Comment:
do the check first before run this async function?

Reply:
async_run_allowing_running_loop has already checked whether there's a event loop in the current thread.

using async here will raise error in linux platform with python3.8/3.9:  RuntimeError: There is no current event loop in thread 'MainThread'.
https://github.com/microsoft/promptflow/pull/1407/checks?check_run_id=19626705439
![image](https://github.com/microsoft/promptflow/assets/26239730/8b8eabc0-68d2-40cb-b8cd-5ed18907b6d8)

Azure core won't use asyncio.Lock() in its latest code and python3.10+ has not supported pass loop in Lock.

Not clear why cause the issue and how to fix it, change from async to threading for now